src/libtiled/tilelayer.h

```diff
@@ -212,6 +212,160 @@ inline const Cell &Chunk::cellAt(const QPoint &point) const
 class TILEDSHARED_EXPORT TileLayer : public Layer
 {
 public:
+    class iterator
+    {
+    public:
+        iterator(QHash<QPoint, Chunk>::iterator it, QHash<QPoint, Chunk>::iterator end)
+            : mChunkPointer(it)
+            , mChunkEndPointer(end)
+        {
+            if (it != end)
+                mCellPointer = it.value().begin();
+        }
+
+        iterator operator++(int)
+        {
+            iterator it = *this;
+
+            if (mChunkPointer != mChunkEndPointer) {
+                if (++mCellPointer == mChunkPointer.value().end()) {
+                    mChunkPointer++;
+                    if (mChunkPointer != mChunkEndPointer)
+                        mCellPointer = mChunkPointer.value().begin();
+                }
+            }
+
+            return it;
+        }
+
+        iterator &operator++()
+        {
+            if (mChunkPointer != mChunkEndPointer) {
+                if (++mCellPointer == mChunkPointer.value().end()) {
+                    mChunkPointer++;
+                    if (mChunkPointer != mChunkEndPointer)
+                        mCellPointer = mChunkPointer.value().begin();
+                }
+            }
```

Comment:
This code is duplicated in the postfix version and I think would be good to separate out in a private `advance` function.